### PR TITLE
[categories] Translation improvements for es

### PR DESCRIPTION
**Pull request by javierllorente**
I have checked the Spanish translation of categories and have found some minor mistakes plus mixture of Latin American and Castilian Spanish. I have corrected them. 
Please note that "es" is now 100% Castilian Spanish.
Greetings!

### CODE REVIEW

**Comment by mpimenov** [comment]
@javierllorente We need you to "sign" the CLA first: https://maps.me/cla/

File: data/categories.txt [hunk 1]
```diff
@@ -757,7 +757,7 @@ pl:^Sklep spożywczy
 pt:^Loja de conveniência
 pt-BR:^Loja de conveniência|4mercadinho|compras
 ro:4^Magazin mixt
-es:^Tienda de abarrotes|chino|bazar
+es:^Tienda|chino|bazar
```
Comment (mpimenov) [inline data/categories.txt]: Are you sure that "chino" and "bazar" are relevant here?
http://wiki.openstreetmap.org/wiki/Tag:shop%3Dconvenience

File: data/categories.txt [hunk 2]
```diff
@@ -822,7 +822,7 @@ pl:^Sklep z elektroniką|sklep elektroniczny
 pt:^Loja de telemóveis|loja de aparelhos eletrónicos
 pt-BR:^Loja de celulares|telefone|compras
 ro:^Magazin de telefoane mobile|magazin de electronice
-es:^Celulares|Tienda de telefonía móvil|tienda de electrónica
+es:^Móviles|Tienda de telefonía móvil|tienda de aparatos electrónicos
```
Comment (mpimenov) [inline data/categories.txt]: Are you sure that "tienda de aparatos electrónicos" is relevant here?
This shop is specifically for mobile phones, after all.

File: data/categories.txt [hunk 3]
```diff
@@ -3652,7 +3653,7 @@ pl:3^Szopa|schronienie
 pt:^Abrigo
 pt-BR:^Abrigo
 ro:^Adăpost
-es:^Refugio|Abrigo|Marquesina
+es:^Refugio|Albergue
```
Comment (mpimenov) [inline data/categories.txt]: Are you sure that "Albergue" is correct?
See http://wiki.openstreetmap.org/wiki/Tag:amenity%3Dshelter

File: data/categories.txt [hunk 4]
```diff
@@ -3718,7 +3719,7 @@ pl:3^Toaleta
 pt:^WC
 pt-BR:^Banheiro
 ro:^Toaletă
-es:^Baños
+es:^WC
```
Comment (mpimenov) [inline data/categories.txt]: Note: it means the primary translation for toilets will now be "WC". Is this really the most used term?

File: data/categories.txt [hunk 5]
```diff
@@ -4428,6 +4429,7 @@ pl:3^Wyjazd|wyjście
 pt:4^Saída|3junção
 pt-BR:2^Saída|3junção
 ro:^Ieșire
+es:^Salida|3intersección
```
Comment (mpimenov) [inline data/categories.txt]: Note: we will start suggesting "intersección" after 3 letters, i.e. after you type "int".
Is this really important enough to provoke a suggestion?

File: data/categories.txt [hunk 6]
```diff
@@ -5297,7 +5299,7 @@ pl:3^Naprawa samochodu|warsztat samochodowy|warsztat|samochód|auto
 pt:^Oficina de reparação de automóveis|estação de serviço
 pt-BR:^Oficina de automóveis|mecânica|mecânico|auto elétro
 ro:^Atelier de reparații auto
-es:^Taller|3reparaciones auto|coche|automóvil|3auto|carro
+es:^Taller|3reparación de automóviles|coche|automóvil
```
Comment (mpimenov) [inline data/categories.txt]: Is "Taller" really the most used term for this?

File: data/categories.txt [hunk 7]
```diff
@@ -7241,7 +7243,7 @@ pl:^Sklep monopolowy
 pt:^Vinho|Loja de Vinhos
 pt-BR:^Loja de vinhos|bebidas
 ro:^Magazin de băuturi alcolice
-es:^Establecimiento de bebidas alcohólicas
+es:^Vino|Tienda de vinos
```
Comment (mpimenov) [inline data/categories.txt]: Note: the "^" term will be used in the UI. Depending on what's actually is written on most signboards, we probably should swap these two synonyms.

File: data/categories.txt (hunk 2 above)
Comment (javierllorente) [reply data/categories.txt]: In this case I have translated "electronics store" from "en"

File: data/categories.txt (hunk 4 above)
Comment (javierllorente) [reply data/categories.txt]: It is very common

File: data/categories.txt (hunk 5 above)
Comment (javierllorente) [reply data/categories.txt]: I have taken the inspiration from "en" and "pt"

File: data/categories.txt (hunk 6 above)
Comment (javierllorente) [reply data/categories.txt]: Yes, it is

**Comment by javierllorente** [comment]
Thank you for going through my changes. I have made some changes and signed the CLA as well

**Pull request merged**